### PR TITLE
docs: consolidate wallet docs

### PR DESCRIPTION
**Pull request by petertonysmith94**
relates: #1505 

### Changes made
- Moved _most_ wallet-based tests from `fuel-gauge` over `docs-snippets`.

### CODE REVIEW

File: packages/fuel-gauge/src/doc-examples.test.ts [hunk 1]
```diff
@@ -152,196 +141,6 @@ describe('Doc Examples', () => {
     expect(arrayify(assetId)).toEqual(arrayify(Address.fromB256(assetId).toB256()));
   });
 
-  test('it can work with wallets', async () => {
```
Comment (Torres-ssf) [inline packages/fuel-gauge/src/doc-examples.test.ts:155]: @petertonysmith94, we do want to move all code snippets to the doc-snippets directory when it is possible. 

However, we cannot remove these e2e test cases from here, as they were most likely added for a specific reason. 

When moving these code snippets, we need to write new, specific tests that validate only the required steps to ensure that the code snippet is working. 

Therefore, in many cases, when migrating a code snippet, many test cases can be simplified.

**Review by Torres-ssf — CHANGES_REQUESTED** [review]
@petertonysmith94 Thanks for helping us with this. We really appreciate it 👍

File: packages/fuel-gauge/src/doc-examples.test.ts (hunk 1 above)
Comment (petertonysmith94) [reply packages/fuel-gauge/src/doc-examples.test.ts:155]: Hey @Torres-ssf, that all sounds reasonable!

I will reinstate the `fuel-gauge` package tests now.

In terms of test simplification, did you have any pointers/guidelines for this process?

From what is there now, I assume that will be generally keeping small, organised, with descriptive test file-name, and having single purpose. Happy to continue with this for now

Comment (Torres-ssf) [reply packages/fuel-gauge/src/doc-examples.test.ts:155]: @petertonysmith94 The conventions that we use for code snippet tests are:

1 - Placing them inside `apps/docs-snippets`

2 - Naming the test file with the same name as the doc page.

3 - Design the test case based on what the code snippets are intended to show.

4- If the code snippet is from a Sway project, it needs to come from one of the projects within `app/docs-snippets/test/fixtures/forc-projects/` or a new one created on this dir.

For instance, on the [Address section](https://fuellabs.github.io/fuels-ts/guide/types/address.html#from-a-public-key), [this specific](https://github.com/FuelLabs/fuels-ts/blob/master/apps/docs-snippets/src/guide/types/address.test.ts#L20-L28) snippet was created to show how to instantiate a wallet from a public key.

**Comment by petertonysmith94** [comment]
Closed in favour of #1885

**Pull request closed without merging**